src/idpi/operators/vertical_interpolation.py

```diff
@@ -253,36 +252,31 @@ def interpolate_k2theta(
     )
 
     # Check that typeOfLevel is supported and equal for field, th_field, and h_field
-    if supported_vc_type not in field.dims:
+    if field.vcoord_type != "model_level" or field.origin["z"] != 0.0:
```

Comment:
what vcord type has a 0-origin and ist not model_level?

And if they are independent, what is the relevant test here?

Reply:
Here we want to raise if the field is not defined on the masspoints, i.e. model level and origin z = 0.

Reply:
so if we want to check for definition on mass points, why is model_level a relevant check? isn't just the origin the relevant part here?

Reply:
The mass points are located on model levels. Fields that are defined on other vertical coordinates are therefore not located on the mass points.

Reply:
in principle, we only have staggered data on model levels, so vcoord_type ="ml". 
All data (I think) we generate (with Fieldextra) on theta or pressure levels is always colocated. 
However, after looking into the grib type of vertical coordinate systems: 
https://github.com/ecmwf/eccodes/blob/6efa9fc52899863d32311e5498f79abde8f303f3/definitions/grib2/typeOfLevelConcept.def#L34
We see that it is possible to have layer on theta and pressure levels. So probably it would make sense to keep the origin defined also for those vcoord_types (even if we do not use them at the moment) ?